utils/docker/vcluster/daos-cm.sh

```diff
@@ -135,31 +135,53 @@ function stop
 
 function start
 {
-	DAOS_IFACE_IP="$1"
-	DAOS_POOL_SIZE="$2"
+	DAOS_IFACE_IP="${1:?Network Interface IP has to be defined}"
+	DAOS_POOL_SIZE="${2:?Pool size has to be defined}"
 
 	info "Starting DAOS virtual cluster containers"
 	if ! run env DAOS_IFACE_IP="$DAOS_IFACE_IP" docker compose up --detach daos_server daos_admin daos_client ; then
 		fatal "DAOS virtual cluster containers could no be started"
 	fi
 
 	info "Waiting for daos-server services to be started"
-	timeout_counter=0
-	until [[ $timeout_counter -ge 5 ]] || run docker exec daos-server systemctl --quiet is-active daos_server ; do
-		info "daos-server not yet ready: waiting 1s"
+	timeout_counter=5
+	until [[ $timeout_counter -le 0 ]] || docker exec daos-server systemctl --quiet is-active daos_server > /dev/null 2>&1 ; do
+		info "daos-server not yet ready: timeout=$timeout_counter"
 		sleep 1
-		(( timeout_counter++ ))
+		(( timeout_counter-- ))
```

Comment:
You need to be careful with `((…))` as it can return !0:

```bash
$ ((1)) && echo true
true
$ ((0)) && echo true
$ ((0)) || echo false
false
```

You probably don't want your script throwing an error simply because the result of your arithmetic operation is 0.

There is an [open issue on Shellcheck](https://github.com/koalaman/shellcheck/issues/1830) about this, FWIW.  And a hint about it in [SC2219](https://www.shellcheck.net/wiki/SC2219).

Ultimately, unless you really do want to actually evaluate the result of your arithmetic operation (beyond just it's action on the variable involved) you will want to append `|| true` to your operation.  I find this to be the case 99% of the time.

Reply:
In this specific case, a value of zero is an error, but it is not intended to be managed this way: timeout of 0 should be handle at line 153.
- [ ] Not return error with counter decrement

Reply:
> In this specific case, a value of zero is an error,

But it would be an _unhandled error_.  That is, it would trigger an exit of this shell right at this line due to the `set -e`.

> but it is not intended to be managed this way: timeout of 0 should be handle at line 153.

Right.  Which will never happen when `$timeout_counter` gets to `0` as the script will exit before that can happen.

One alternative (although I'm not sure I like it as much as the existing structure if you added `|| true` to this line is to do this:

```patch
-	until [[ $timeout_counter -le 0 ]] || docker exec daos-server systemctl --quiet is-active daos_server > /dev/null 2>&1 ; do
+	until docker exec daos-server systemctl --quiet is-active daos_server > /dev/null 2>&1 ; do
 		info "daos-server not yet ready: timeout=$timeout_counter"
 		sleep 1
-		(( timeout_counter-- ))
+		if ! (( timeout_counter-- )); then
+			fatal "DAOS server could not be started"
+		fi
	done
-	if [[ $timeout_counter -le 0 ]] ; then
-		fatal "DAOS server could not be started"
-	fi
```

Reply:
Like this :)

Reply:
Fixed with commit e25d91c6d2fa642272df16b54ab50b2e221eb470